nima/http/media/jsonp/src/main/java/io/helidon/nima/http/media/jsonp/JsonpMediaSupportProvider.java

```diff
@@ -16,127 +16,38 @@
-        return new JsonpWriter<>(WRITER_FACTORY);
-    }
+public class JsonpMediaSupportProvider implements MediaSupportProvider, Weighted {
 
     /**
-     * Server request reader direct access.
-     *
-     * @param <T> type to read
-     * @return a reader to read JSON-P objects
+     * This class should be only instantiated as part of java {@link java.util.ServiceLoader}.
      */
-    public static <T extends JsonStructure> EntityReader<T> serverRequestReader() {
-        return new JsonpReader<>(READER_FACTORY);
+    @Deprecated
+    public JsonpMediaSupportProvider() {
+        super();
     }
 
     @Override
-    public <T> ReaderResponse<T> reader(GenericType<T> type, Headers requestHeaders) {
-        if (isSupportedType(type)) {
-            if (requestHeaders.contentType()
-                    .map(it -> it.test(MediaTypes.APPLICATION_JSON))
-                    .orElse(true)) {
-                return new ReaderResponse<>(SupportLevel.SUPPORTED, this::reader);
-            }
-        }
-        return ReaderResponse.unsupported();
+    public String configKey() {
+        return "jsonp";
     }
 
     @Override
-    public <T> WriterResponse<T> writer(GenericType<T> type,
-                                        Headers requestHeaders,
-                                        WritableHeaders<?> responseHeaders) {
-
-        if (isSupportedType(type)) {
-            return new WriterResponse<>(SupportLevel.SUPPORTED, this::writer);
-        }
-
-        return WriterResponse.unsupported();
-    }
-
-    @Override
-    public <T> ReaderResponse<T> reader(GenericType<T> type,
-                                        Headers requestHeaders,
-                                        Headers responseHeaders) {
-        if (isSupportedType(type)) {
-            List<HttpMediaType> acceptedTypes = requestHeaders.acceptedTypes();
-            // check if accepted
-            if (acceptedTypes.isEmpty()) {
-                return new ReaderResponse<>(SupportLevel.SUPPORTED, this::reader);
-            }
-            for (HttpMediaType acceptedType : acceptedTypes) {
-                if (acceptedType.test(MediaTypes.APPLICATION_JSON)) {
-                    return new ReaderResponse<>(SupportLevel.SUPPORTED, this::reader);
-                }
-            }
-        }
-
-        return ReaderResponse.unsupported();
+    public MediaSupport create(Config config) {
+        return JsonpSupport.create(config);
     }
 
     @Override
-    public <T> WriterResponse<T> writer(GenericType<T> type, WritableHeaders<?> requestHeaders) {
-        // if the type is JsonStructure, we support it and do not care about content type
-        // you provide json, you get json...
-        if (isSupportedType(type)) {
-            return new WriterResponse<>(SupportLevel.SUPPORTED, this::writer);
-        }
-        return WriterResponse.unsupported();
+    public double weight() {
```

Comment:
JSON-P has to have higher than both JSON-B and Jackson, but it does not need to be low, as it supports very specific types (`JsonObject`, `JsonArray` etc.)

Reply:
OK. As I understand all the notes here and in the code, weights shall be JSON-P > Jackson > JsonB
and JSON-P shall be high, while Jackson and JsonB very low.
It's set to:

- JsonbMediaSupportProvider - 10
- JacksonMediaSupportProvider - 15
- JsonpMediaSupportProvider - 1000